libcudacxx/include/cuda/std/__complex/std_complex_access.h

```diff
@@ -0,0 +1,84 @@
+#ifndef _LIBCUDACXX___COMPLEX_STD_COMPLEX_ACCESS_H
+#define _LIBCUDACXX___COMPLEX_STD_COMPLEX_ACCESS_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if !_CCCL_COMPILER(NVRTC)
+
+#  include <cuda/std/__type_traits/is_constant_evaluated.h>
+
+#  include <complex>
+
+#  if !defined(__CUDA_ARCH__) || defined(_CCCL_BUILTIN_IS_CONSTANT_EVALUATED)
+#    define _LIBCUDACXX_CONSTEXPR_STD_COMPLEX_ACCESS       constexpr
+#    define _LIBCUDACXX_HAS_CONSTEXPR_STD_COMPLEX_ACCESS() 1
+#  else // ^^^ _CCCL_BUILTIN_IS_CONSTANT_EVALUATED ^^^ / vvv !_CCCL_BUILTIN_IS_CONSTANT_EVALUATED vvv
+#    define _LIBCUDACXX_CONSTEXPR_STD_COMPLEX_ACCESS
+#    define _LIBCUDACXX_HAS_CONSTEXPR_STD_COMPLEX_ACCESS() 0
+#  endif // ^^^ !_CCCL_BUILTIN_IS_CONSTANT_EVALUATED ^^^
```

Comment:
I believe we can just mark them `constexpr` if we restrict ourself to host only operations